rsocket-transport-netty/src/main/java/io/rsocket/transport/netty/server/WebsocketServerTransport.java

```diff
@@ -147,27 +161,92 @@ public void setTransportHeaders(Supplier<Map<String, String>> transportHeaders)
     Objects.requireNonNull(acceptor, "acceptor must not be null");
 
     Mono<CloseableChannel> isError = FragmentationDuplexConnection.checkMtu(mtu);
-    return isError != null
-        ? isError
-        : server
-            .handle(
-                (request, response) -> {
-                  transportHeaders.get().forEach(response::addHeader);
-                  return response.sendWebsocket(
-                      null,
-                      FRAME_LENGTH_MASK,
-                      (in, out) -> {
-                        DuplexConnection connection =
-                            new WebsocketDuplexConnection((Connection) in);
-                        if (mtu > 0) {
-                          connection =
-                              new FragmentationDuplexConnection(
-                                  connection, ByteBufAllocator.DEFAULT, mtu, false, "server");
-                        }
-                        return acceptor.apply(connection).then(out.neverComplete());
-                      });
-                })
-            .bind()
-            .map(CloseableChannel::new);
+
+    if (isError != null) {
+      return isError;
+    }
+
+    if (headersPredicate != null) {
```

Comment:
Couldn't this be simplified by having a default headers predicate that always returns true?